lib/portage/dbapi/porttree.py

```diff
@@ -1351,14 +1343,15 @@ async def async_xmatch(self, level, origdep, loop=None):
                 except InvalidData:
                     continue
 
-                if visibility_filter and not self._visible(pkg_str, metadata):
-                    continue
-
-                if mydep.slot is not None and not _match_slot(mydep, pkg_str):
-                    continue
-
-                if mydep.unevaluated_atom.use is not None and not self._match_use(
-                    mydep, pkg_str, metadata, ignore_profile=ignore_profile
+                if (
+                    visibility_filter
+                    and not self._visible(pkg_str, metadata)
+                    or mydep.slot is not None
+                    and not _match_slot(mydep, pkg_str)
+                    or mydep.unevaluated_atom.use is not None
+                    and not self._match_use(
+                        mydep, pkg_str, metadata, ignore_profile=ignore_profile
+                    )
```

Comment:
Does this have the same logical effect of the original? Maybe add parenthesis around the original independent conditions so it's more readable that any of these conditions result in success.

Or maybe even an `anyof()`.

Reply:
I love the `any()` idea and using that to implement the requested change. 😎